tests/Unit/CrudPanel/CrudPanelColumnsTest.php

```diff
@@ -0,0 +1,242 @@
+<?php
+
+namespace Backpack\CRUD\Tests\Unit\CrudPanel;
+
+class CrudPanelColumnsTest extends BaseCrudPanelTest
+{
+    private $oneColumnArray = [
+        'name' => 'column1',
+        'label' => 'Column1',
+    ];
+
+    private $otherOneColumnArray = [
+        'name' => 'column4',
+        'label' => 'Column4',
+    ];
+
+    private $twoColumnsArray = [
+        [
+            'name' => 'column1',
+            'label' => 'Column1',
+        ],
+        [
+            'name' => 'column2',
+            'label' => 'Column2',
+        ],
+    ];
+
+    private $expectedTwoColumnsArray = [
+        'column1' => [
+            'name' => 'column1',
+            'label' => 'Column1',
+        ],
+        'column2' => [
+            'name' => 'column2',
+            'label' => 'Column2',
+        ],
+    ];
+
+    private $threeColumnsArray = [
+        [
+            'name' => 'column1',
+            'label' => 'Column1',
+        ],
+        [
+            'name' => 'column2',
+            'label' => 'Column2',
+        ],
+        [
+            'name' => 'column3',
+            'label' => 'Column3',
+        ],
+    ];
+
+    private $expectedThreeColumnsArray = [
+        'column1' => [
+            'name' => 'column1',
+            'label' => 'Column1',
+        ],
+        'column2' => [
+            'name' => 'column2',
+            'label' => 'Column2',
+        ],
+        'column3' => [
+            'name' => 'column3',
+            'label' => 'Column3',
+        ],
+    ];
+
+    public function testAddColumnByName()
+    {
+        $this->crudPanel->addColumn('column1');
+
+        $this->assertEquals(1, count($this->crudPanel->columns));
+        $this->assertContains($this->oneColumnArray, $this->crudPanel->columns);
+    }
+
+    public function testAddColumnsByName()
+    {
+        $this->crudPanel->addColumns(['column1', 'column2']);
+
+        $this->assertEquals(2, count($this->crudPanel->columns));
+        $this->assertEquals($this->expectedTwoColumnsArray, $this->crudPanel->columns);
+    }
+
+    public function testAddColumnAsArray()
+    {
+        $this->crudPanel->addColumn($this->oneColumnArray);
+
+        $this->assertEquals(1, count($this->crudPanel->columns));
+        $this->assertContains($this->oneColumnArray, $this->crudPanel->columns);
+    }
+
+    public function testAddColumnsAsArray()
+    {
+        $this->crudPanel->addColumns($this->twoColumnsArray);
+
+        $this->assertEquals(2, count($this->crudPanel->columns));
+        $this->assertEquals($this->expectedTwoColumnsArray, $this->crudPanel->columns);
+    }
+
+    public function testAddColumnNotArray()
+    {
+        $this->setExpectedException(\ErrorException::class);
+
+        $this->crudPanel->addColumns('column1');
```

Comment:
Is this missing an assert?

Reply:
Nope, this test method expects an exception. If the exception is not thrown, the test method will fail.

Reply:
Yep, sorry was speed reviewing half asleep.